utils/yoke/src/yoke.rs

```diff
@@ -74,14 +74,25 @@ use alloc::sync::Arc;
 /// assert_eq!(&**yoke.get(), "hello");
 /// assert!(matches!(yoke.get(), &Cow::Borrowed(_)));
 /// ```
-#[derive(Debug)]
 pub struct Yoke<Y: for<'a> Yokeable<'a>, C> {
     // must be the first field for drop order
     // this will have a 'static lifetime parameter, that parameter is a lie
     yokeable: Y,
     cart: C,
 }
 
+impl<Y: for<'a> Yokeable<'a>, C: core::fmt::Debug> core::fmt::Debug for Yoke<Y, C>
+where
+    for<'a> <Y as Yokeable<'a>>::Output: core::fmt::Debug,
```

Comment:
Please write a unit test for this because we've had issues in the past with these `for<'a>` bounds not compiling at the call sites.

Reply:
I’ve added a simple test. Is that the kind of test you were thinking of? Of course a simple test does not cover all cases; maybe there’s something with more complicated lifetimes that _does_ cause problems even though it shouldn’t :thinking: If you know any concrete past issues with HRTBs you can link to so I can see if there’s a better test case to extract from that.

Reply:
Ah… is this related to why [`YokeTraitHack`](https://docs.rs/yoke/latest/yoke/trait_hack/struct.YokeTraitHack.html) exists? I’ll look into that…

Reply:
@Manishearth since you’re the one who opened https://github.com/rust-lang/rust/issues/89196, maybe you can comment on this? Should this be re-written to use the `YokeTraitHack` type, similar to how the `Clone` implementation looks? Where can I find a reproducing example of the issue using the actual `yoke` crate’s API?

Reply:
I'm happy to avoid the trait hack as long as this _works_ , and if it stops working we can revisit the trait hack. Tha'ts been my general attitude towards it :smile:

Reply:
Alright, got it :-)